### DISCUSSION TITLE
HTML tag can break the pagination

### DISCUSSION THREAD

**Discussion by rabol** [livewire/livewire#3498] [7. Bugs (archived)]
Hi

Adding some HTML to a component can break the pagination

Steps:
create a new laravel app
add laravel ui, and use bootstrap

Create a new comonent:

```
?php

namespace App\Http\Livewire;

use App\Models\Post;
use Livewire\Component;
use Livewire\WithPagination;

class Posts extends Component
{
    use WithPagination;

    protected $paginationTheme = 'bootstrap';

    public function render()
    {
        return view('livewire.posts')->with('posts', Post::paginate(10));
    }
}
```

and the view:

```
{{-- uncomment to make the pagination work
<h4>Test</h4> 
 --}}
    <div>
        <table class="table" style="width: 100%;">
            <thead>
                <tr>
                    <th class="">ID</th>
                    <th class="">Title</th>

                </tr>
            </thead>
            <tbody>
                @foreach ($posts as $post)
                    <tr>
                        <td class="border px-4 py-2">{{ $post->id }}</td>
                        <td class="border px-4 py-2">{{ $post->title }}</td>

                    </tr>
                @endforeach
            </tbody>
        </table>

        {{ $posts->links() }}
    </div>

```

use tinker to create some records

display the component and see that it works
Then uncomment the first line in the view and now the pagination does not work

**Comment by ryatkins** [comment]
Livewire components rely on having an outer `div` to do the DOM diffing for the component.  By adding something outside your most outer `div` you're most likely breaking the diffing engine.  Wrap your whole component in another outer `div` and it should work.

**Comment by joshhanley** [comment]
@rabol yeah @ryatkins is correct, you should only have one root element in your component. It doesn't explicitly have to be a div, but its generally the best option.

From the docs https://laravel-livewire.com/docs/2.x/rendering-components#returning-blade

>Make sure your Blade view only has ONE root element.

Hope this helps!
